### PR TITLE
Change grab cursors; fix some bugs

### LINKED ISSUE #10166: Add hand to drag cursor (part 2)
Red zone: 
- Hover: Open hand
- Mouse down: Closed hand

Blue zone:
- Hover: Click hand (as today, but zone is bigger)

Green zone:
- Hover: Text edit cursor

Unshaded:
- Keep existing behaviour

image

image

image

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Change grab cursors:
- Node has grab/grabbed icons for only halo and icon.

Fix bugs:
- Empty part of top bar no longer blocks mouse events.
- JSON viz: Clickable parts of inline elements now use pointer cursor when hovered.
- Doc panel breadcrumbs: Icon can be clicked (behavior now consistent with cursor shown).

https://github.com/enso-org/enso/assets/1047859/3e48a6c1-3f43-497f-82ad-eb787e9c9643

Closes #10166.

### Important Notes

- New global `clickable` class replaces `cursor: pointer`; the class can be applied closer to the event handler.
- Refactor: Extracted `GraphNode` output port logic to a new component.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: app/gui2/src/assets/base.css [hunk 1]
```diff
@@ -57,6 +57,11 @@
   --right-dock-default-width: 40%;
 }
 
+/* Constants */
+:root {
```
Comment (MrFlashAccount) [inline app/gui2/src/assets/base.css]: Not sure we need the `--pi` variable here: 
https://developer.mozilla.org/en-US/docs/Web/CSS/calc-constant#pi

File: app/gui2/src/components/GraphEditor/GraphNode.vue [hunk 2]
```diff
@@ -845,11 +683,7 @@ watchEffect(() => {
   margin-right: 4px;
 }
 
-.draggable {
-  cursor: grab;
-}
-
 .dragged {
```
Comment (MrFlashAccount) [inline app/gui2/src/components/GraphEditor/GraphNode.vue:682]: When we want to apply draggable styles when a user drags an item, there are some caveats we need to consider:

1. Element can't keep up the mouse, eg when render takes more than 16ms
2. User drags away the draggable element

So it's better to either apply draggable to the whole document, or use something like 

```css
body:has(.dragged) {
   cursor: grabbing !important;
}
```


https://github.com/enso-org/enso/assets/61194245/53905526-1737-48ab-923c-47fefd6e7adf

File: app/gui2/src/components/GraphEditor/widgets/WidgetText.vue [hunk 3]
```diff
@@ -99,7 +99,6 @@ export const widgetDefinition = defineWidget(
       ref="input"
       v-model="editedContents"
       autoSelect
-      @pointerdown.stop
```
Comment (farmaazon) [inline app/gui2/src/components/GraphEditor/widgets/WidgetText.vue]: Why is this removed? The pointerdown are actually handled by input (pointerdown, not click sets the text cursor), thence it should be also stopped here.

File: app/gui2/src/components/widgets/DropdownWidget.vue [hunk 4]
```diff
@@ -63,21 +63,22 @@ export interface DropdownEntry {
...
-      </template>
+      <li
+        v-for="entry in sortedValues"
+        :key="entry.value"
+        :class="{ selected: entry.selected }"
+        class="item clickable"
+        @click.stop="emit('clickEntry', entry, $event.altKey)"
+      >
+        <div v-if="entry.selected"><span v-text="entry.value"></span></div>
+        <span v-else v-text="entry.value"></span>
```
Comment (farmaazon) [inline app/gui2/src/components/widgets/DropdownWidget.vue]: Do we need this <div> when item is selected? The div does not seem to have any styling attached (in contrast to previous implementation).

File: app/gui2/src/components/GraphEditor/GraphNode.vue (hunk 2 above)
Comment (kazcw) [reply app/gui2/src/components/GraphEditor/GraphNode.vue:682]: We are using `setPointerCapture`, so the cursor style doesn't depend on what node is under the pointer. I think it is enough to apply `dragged` to the draggable elements.

Comment (MrFlashAccount) [reply app/gui2/src/components/GraphEditor/GraphNode.vue:682]: TIL!

**Review by farmaazon — APPROVED** [review]
